client.go

```diff
@@ -155,6 +156,10 @@ func (c *Client) Connect(ctx context.Context) (err error) {
 	return nil
 }
 
+func (c *Client) sessionClosed() bool {
+	return c.Session() != nil
```

Comment:
Probably not! It does seems odd indeed, It's certainly a place holder for a function that was meant to be implemented later on and I must have forgotten to write a TODO comment for myself.
At the moment I am getting familiar with this code since It's been a while that I have last lay eyes on it...
Anyway thanks for your remark I'll get to the bottom of it, and sorry for the sloppiness!